packages/pn-personafisica-webapp/src/utils/delegation.utility.ts

```diff
@@ -28,31 +31,52 @@ export default function delegationToItem(
   }));
 }
 
+export const getEmailFromDelegation = (delegation: Delegation): string => {
+  if ('delegator' in delegation && delegation.delegator) {
+    return delegation.delegator.email;
+  }
+  if ('delegate' in delegation && delegation.delegate) {
+    return delegation.delegate.email;
+  }
+  return '';
+};
+
 export function generateVCode() {
   const crypto = window.crypto;
   const array = new Uint32Array(1);
   return crypto.getRandomValues(array).toString().slice(0, 5);
 }
 
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
@AndriBoca nota metodologica: se in una class o una function c'e' una complessita' cognitiva elevata, sforzarsi di rifattorizzare per ridurla. I commenti di disabilitazione della regola possono essere usati solo temporaneamente durante le fasi di sviluppo. Per ora rifattorizzo io questo metodo, ti chiederei di farne una analisi a valle del refactoring che faro'.